lib/vanagon/component/source/http.rb

```diff
@@ -121,24 +116,20 @@ def extract(tar)
         def cleanup
           if ARCHIVE_EXTENSIONS.include?(@extension)
             return "rm #{@file}; rm -r #{dirname}"
-          elsif NON_ARCHIVE_EXTENSIONS.include?(@extension)
+          else
             # Because dirname will be ./ here, we don't want to try to nuke it
             return "rm #{@file}"
-          else
-            fail "Don't know how to cleanup for '#{@file}' with extension: '#{@extension}'. Please teach me."
           end
         end
 
         # Returns the extension for @file
         #
         # @return [String] the extension of @file
-        # @raise [RuntimeError] an exception is raised if the extension isn't in the current list
         def get_extension
-          extension_match = @file.match(/.*(#{Regexp.union(ARCHIVE_EXTENSIONS + NON_ARCHIVE_EXTENSIONS)})/)
+          extension_match = @file.match(/.*(#{Regexp.union(ARCHIVE_EXTENSIONS)})/)
           unless extension_match
-            fail "Unrecognized extension for '#{@file}'. Don't know how to extract this format. Please teach me."
+            return @file.split('.').last
```

Comment:
Are there any plaintext files with more than one `.` in the extension?

Also, I think this will return `txt` instead of `.txt`, which used to be returned.